goal_src/jak2/levels/city/common/nav-graph-h.gc

```diff
@@ -7,3 +7,224 @@
+   (src-node          nav-node   :offset          0)
+   (dest-node         nav-node   :offset          4)
+   (temp-dest-node-id int32      :offset          4)
+   (speed-limit       uint8      :offset-assert   8)
+   (density           uint8      :offset-assert   9)
+   (clock-type        uint8      :offset-assert  10)
+   (clock-mask        uint8      :offset-assert  11)
+   (max-user-count    uint8      :offset-assert  12)
+   (user-count        uint8      :offset-assert  13)
+   (width             uint8      :offset-assert  14)
+   (flags             uint8      :offset-assert  15)
+   )
+  :method-count-assert 21
+  :size-assert         #x10
+  :flag-assert         #x1500000010
+  (:methods
+    (nav-branch-method-9 () none 9)
+    (nav-branch-method-10 () none 10)
+    (nav-branch-method-11 (_type_) float 11)
+    (nav-branch-method-12 (_type_) float 12)
+    (nav-branch-method-13 (_type_) float 13)
+    (user-limit-reached? (_type_) symbol 14)
+    (nav-branch-method-15 (_type_) symbol 15)
+    (nav-branch-method-16 () none 16)
+    (nav-branch-method-17 () none 17)
+    (nav-branch-method-18 () none 18)
+    (nav-branch-method-19 () none 19)
+    (nav-branch-method-20 () none 20)
+    )
+  )
+
+
+(deftype nav-node (structure)
+  ((data         uint32             8       :offset-assert   0)
+   (position     vector             :inline :offset          0)
+   (pos-x        float                      :offset          0)
+   (pos-y        float                      :offset          4)
+   (pos-z        float                      :offset          8)
+   (angle        uint16                     :offset         12)
+   (id           uint16                     :offset         14)
+   (radius       uint8                      :offset         16)
+   (branch-count int8                       :offset         17)
+   (flags        nav-node-flag-byte         :offset         18)
+   (pad0         int8               1       :offset         19)
+   (branch-array uint32                     :offset         20)
+   (nav-mesh-id  uint32                     :offset         24)
+   (level        level                      :offset         28)
+   )
+  :method-count-assert 22
+  :size-assert         #x20
+  :flag-assert         #x1600000020
+  (:methods
+    (nav-node-method-9 () none 9)
+    (nav-node-method-10 () none 10)
+    (nav-node-method-11 () none 11)
+    (nav-node-method-12 () none 12)
+    (nav-node-method-13 () none 13)
+    (nav-node-method-14 () none 14)
+    (nav-node-method-15 () none 15)
+    (nav-node-method-16 () none 16)
+    (nav-node-method-17 () none 17)
+    (get-position (_type_ vector) vector 18)
+    (calc-sine-and-cosine! (_type_ vector) vector 19)
+    (get-angle (_type_) float 20)
+    (nav-node-method-21 (_type_) float 21)
+    )
+  )
+
+
+(defmethod nav-branch-method-11 nav-branch ((obj nav-branch))
+  "TODO
+      @returns `density * 0.0078125` - is this some kind of trick?"
+  (* 0.0078125 (the float (-> obj density)))
```

Comment:
I think this method (and the one below) are just to help make the `nav-branch`s smaller in memory.

They store the density as an `int8`, and this rescales that value to some more meaningful scale.